code/modules/power/tesla/generator.dm

```diff
@@ -3,4 +3,7 @@
 	desc = "Makes the wardenclyffe look like a child's plaything when shot with a particle accelerator."
 	icon = 'icons/obj/tesla_engine/tesla_generator.dmi'
 	icon_state = "TheSingGen"
-	creation_type = /obj/singularity/energy_ball
\ No newline at end of file
+	creation_type = /obj/singularity/energy_ball
+
+/obj/machinery/the_singularitygen/tesla/tesla_act(power, explosive = FALSE)
```

Comment:
well this won't really do anything since the only thing this procs does is set the energy